tests/browser_context_test.go

```diff
@@ -704,17 +707,18 @@ func TestNewTab(t *testing.T) {
 	mux.Handle(path, http.StripPrefix(path, fs))
 
 	// Start the iteration
-	_, rt, _, cleanUp := startIteration(t, env.ConstLookup(env.K6TestRunID, "12345"))
+	vu, _, _, cleanUp := startIteration(t, env.ConstLookup(env.K6TestRunID, "12345"))
 	defer cleanUp()
 
 	// Run the test script
-	_, err := rt.RunString(fmt.Sprintf(`
-		const p = browser.newPage()
-		p.goto("%s/%s/ping.html")
-		
-		const p2 = browser.context().newPage()
-		p2.goto("%s/%s/ping.html")
-	`, s.URL, testBrowserStaticDir, s.URL, testBrowserStaticDir))
+	_, err := vu.TestRT.RunOnEventLoop(fmt.Sprintf(`
+		(async function() {
+			const p = await browser.newPage()
+			p.goto("%s/%s/ping.html")
```

Comment:
And in here 😅

Reply:
Ah, nice catch 👏 I'll send another PR for this to not to mess with the chain of other async PRs 🥲